reconcile/gitlab_housekeeping.py

```diff
@@ -202,19 +203,41 @@ def is_rebased(mr, gl: GitLabApi) -> bool:
 
 
 def get_labels(mr: ProjectMergeRequest, gl: GitLabApi) -> list[str]:
-    labels = mr.attributes.get("labels")
-    if not labels:
-        # Sometimes the label attribute is empty but shouldn't. Try it again by fetching this MR separately
-        labels = gl.get_merge_request_labels(mr.iid)
-    return labels
+    """
+    This function used to contain logic for checking if labels were empty and calling
+    gl.get_merge_request_labels() if they were missing because it was suggested that the
+    label attribute is empty sometimes when it shouldn't be. This was an expensive
+    approach to what doesn't appear like it's a consistent problem. We can revisit in
+    the future if we find that this is problematic.
+    """
+    return mr.attributes.get("labels")
+
+# The cache methods below are a quick solution to a problem that probably would
+# otherwise result in more refactoring. We're calling the GitLab API multiple times
+# for the same data over the course of a single integration run.
+
+
+@cache
+def _get_merge_requests(gl: GitLabApi):
+    return gl.get_merge_requests(state=MRState.OPENED)
+
+
+@cache
+def _get_merge_request_commits(mr: ProjectMergeRequest):
+    return mr.commits()
+
+
+@cache
+def _get_merge_requst_label_events(gl: GitLabApi, mr: ProjectMergeRequest):
+    return gl.get_merge_request_label_events(mr)
```

Comment:
Please add type hints for the return values of the methods.

Reply:
Agreed, this is still in draft, and the return of the methods that I'm wrapping don't have type hints either, so I figured I'd defer while waiting to see if anyone had any obvious feedback why this was a bad idea. I'll dig a bit more to make sure that there aren't any issues with stale data and add hints.